pytorch_lightning/profiler/base.py

```diff
@@ -64,6 +71,18 @@ def __init__(
         self._local_rank: Optional[int] = None
         self._stage: Optional[str] = None
 
+    @abstractmethod
```

Comment:
what is the benefit of changing these to abstractmethods?

Reply:
I think it more clearly indicates to users that they need to be implemented. Simply having a NotImplementedError is easier to miss in my opinion

Reply:
sg!